pkg/tests/ossm/smoke_test.go

```diff
@@ -63,6 +132,56 @@ func TestBookinfoInjection(t *testing.T) {
 	})
 }
 
+func validateStartUpProxyTime(t TestHelper, ns string) {
+	podTimes := ExtractProxyTimes(t, ns)
+	for _, podTime := range podTimes {
+		startupTime := podTime.LastTransitionTime.Sub(podTime.StartedAt)
+		t.Logf("Proxy startup time: %s", startupTime.String())
+		if startupTime > 10*time.Second {
+			t.Fatalf("Proxy startup time is too long: %s", startupTime.String())
+		}
+	}
+}
+
+func ExtractProxyTimes(t TestHelper, ns string) []PodTime {
+	t.Log("Extracting proxy startup time and last transition time for all the pods in the namespace")
+	proxyTimeList := oc.GetJson(t, ns, "pods", "", `{range .items[*]}{.metadata.name}{"\t"}{.status.containerStatuses[?(@.name=="istio-proxy")].state.running.startedAt}{"\t"}{.status.conditions[?(@.type=="Ready")].lastTransitionTime}{"\n"}{end}`)
+
+	var podTimes []PodTime
+
+	scanner := bufio.NewScanner(strings.NewReader(proxyTimeList))
+	for scanner.Scan() {
+		line := scanner.Text()
+		fields := strings.Fields(line)
+		podName := fields[0]
+
+		// Skip if we have less than 3 fields, because we expect 3 fields per line
+		if len(fields) < 3 {
+			continue
+		}
```

Comment:
Does this ever happen?

Reply:
Yes, because in bookinfo namespace there is a sleep pod without proxy so the length of that line is 2